### DISCUSSION TITLE
Project architecture: Separate "backend" and "frontend"

### DISCUSSION THREAD

**Discussion by fefzdev** [ecvdbdx/git-view#47] [Ideas]
We currently have all sides of the app at the same level of architecture. The views of the app are mixed up with the Electron backend logic and there is no clear boundaries to separate them. 
I'm not sure that create `frontend/` and `backend/` folders is a good solution but I think we should think about a better way to organise our app.

### Pros:
* More logical folder architecture
* Separate app services 
### Cons:
* Might need to refactor

**Comment by MatthieuTD** [comment]
It's a good idea, it will allow you to review part of the app to know if everything is in the right place (refracto)

**Reply by fefzdev** [reply]
Agreed! We should separate Electron logic from the reste of the app

**Comment by LahaxeHugo** [comment]
It might be a lot of work but it while be easier to expand in the future.

**Reply by fefzdev** [reply]
Yup it will need a dedicated ticket imo. The app isn't that big for now so the sooner we tackle this the easier it will be

**Comment by fefzdev** [comment]
I'm not sure about how we could organize it 🤔  I'm open to suggestions

**Reply by fefzdev** [reply]
```
src/
├─ backend/
├─ frontend/
```
isn't a good answer imo

**Reply by fefzdev** [reply]
I'm using this to generate the tree btw
[https://ascii-tree-generator.com/](https://ascii-tree-generator.com/)

**Comment by fefzdev** [comment]
I'm wondering that maybe the issue is just the naming and pattern of the `services/` folder. Files in it are closest to APIs talking with the terminal and OS folders than services. Moreover, the `services/` folder is kind of the only folder which be designed as "backend"
```
api/
├─ gitReader/
│  ├─ __tests__/
│  ├─ index.js
├─ folderReader/
│  ├─ __tests__/
│  ├─ index.js
├─ index.js
```
With a folder structure like this we could add other utils folders like constants or helpers in its root or even more specific ones in each API "services".
```
api/
├─ utils/
│  ├─ constants/
│  ├─ helpers/
├─ gitReader/
├─ folderReader/
│  ├─ constants/
│  ├─ helpers/
│  ...
├─ index.js
```

I'm also of the opinion of dropping the class structure in these files since we are only using it there and I find it a bit verbose for no reason. 
RFC
